artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/ActiveMQChannelHandler.java

```diff
@@ -104,6 +105,7 @@ public void exceptionCaught(final ChannelHandlerContext ctx, final Throwable cau
       // and we don't want to spew out stack traces in that event
       // The user has access to this exeception anyway via the ActiveMQException initial cause
 
+      exception = true;
```

Comment:
I downloaded the branch, and this is basically not effective.

exceptionCaught (the method where this is being set) will return if !active. Which means this change here is pointless.


that's why I think it's important to reproduce the issue on a Leak test.

First: because I had fixed a few open wire leaks.. if this is another leak, it would be better to first reproduce the issue.

Reply:
I believe the point of the report was, this being callled while it _is_ still active, which further down this method then sets active=false, which then meant by the time it later called channelInactive that it didnt fire the connectionDestroyed listener. So this adds the exception=true flag for that case so that exceptionCaught firing whilst active records that, and then when channelInactive fired later it could still fire connectionDestroyed from that.

My earlier feedback was that while that seems reasonable enough at the server side, the Core client side also uses the same handler methods ,and there it looks less appropriate as those are both triggers for failover. Its probably the client usage that influenced the current setup where either exceptionCaught or channelInactive gets to handle 'problem while active' then setting active=false so the other one cant do anything.